### PR TITLE
LF-2472 add a language selector drop down on the account creation information page

### PR DESCRIPTION
**Pull request by seinnlee**
Ticket [LF-2472](https://lite-farm.atlassian.net/browse/LF-2472)

To Test:
1. Create a new account
2. Make sure the language preference selection works (it should detect the browser language and selecting a different language will change the language of the app immediately)

### CODE REVIEW

File: packages/webapp/src/components/CreateUserAccount/index.jsx [hunk 1]
```diff
@@ -22,31 +23,69 @@ export default function PureCreateUserAccount({ onSignUp, email, onGoBack }) {
...
   ];
 
+  const languageOptions = [
+    { value: 'en', label: t('PROFILE.ACCOUNT.ENGLISH') },
+    { value: 'es', label: t('PROFILE.ACCOUNT.SPANISH') },
+    { value: 'pt', label: t('PROFILE.ACCOUNT.PORTUGUESE') },
+    { value: 'fr', label: t('PROFILE.ACCOUNT.FRENCH') },
+  ];
+
+  const getLanguageOption = (language) => {
```
Comment (anant00729) [inline packages/webapp/src/components/CreateUserAccount/index.jsx:58]: can add option as a key in the { value: 'en', label: t('PROFILE.ACCOUNT.ENGLISH') },

e.g { value: 'en', label: t('PROFILE.ACCOUNT.ENGLISH'), option: 0 },

File: packages/webapp/src/components/CreateUserAccount/index.jsx [hunk 2]
```diff
@@ -22,31 +23,69 @@ export default function PureCreateUserAccount({ onSignUp, email, onGoBack }) {
...
+    ? navigator.language.split('-')[0]
+    : navigator.language;
+
+  const [language, setLanguage] = useState(browser_langauge);
+  const [languageOption, setLanguageOption] = useState(getLanguageOption(language));
+
+  useEffect(() => {
+    setLanguageOption(getLanguageOption(language));
+    i18n.changeLanguage(language);
+    localStorage.setItem('litefarm_lang', language);
```
Comment (anant00729) [inline packages/webapp/src/components/CreateUserAccount/index.jsx:81]: can put 'litefarm_lang' in the constants 
e.g 

export const LITEFARM_LANG = 'litefarm_lang'

File: packages/webapp/src/components/CreateUserAccount/index.jsx (hunk 1 above)
Comment (seinnlee) [reply packages/webapp/src/components/CreateUserAccount/index.jsx:58]: Why do we need a key here?

Comment (anant00729) [reply packages/webapp/src/components/CreateUserAccount/index.jsx:58]: const getLanguageOption = (language) => {
    switch (language) {
      case 'en':
        return 0;
      case 'es':
        return 1;
      case 'pt':
        return 2;
      case 'fr':
        return 3;
    }
  };

  you could have tried key value pairs of language 

  e.g: 
getLanguageOption = (language) => {
  const options = {
    'en': 0,
    'es': 1,
    'pt': 2,
    'fr': 2,
  }
  return options[language]
}

Comment (anant00729) [reply packages/webapp/src/components/CreateUserAccount/index.jsx:58]: const languageOptions = [
  { 
    value: 'en', 
    label: t('PROFILE.ACCOUNT.ENGLISH')
  },
  { value: 'es', label: t('PROFILE.ACCOUNT.SPANISH') },
  { value: 'pt', label: t('PROFILE.ACCOUNT.PORTUGUESE') },
  { value: 'fr', label: t('PROFILE.ACCOUNT.FRENCH') },
];


Also, you can use index position to get the values as 0, 1, 2, 3

Comment (anant00729) [reply packages/webapp/src/components/CreateUserAccount/index.jsx:58]: const languageOptions = [
{ value: 'en', label: 'english'},
{ value: 'es', label: 'spanish' },
{ value: 'pt', label: 'Portuguese ' },
{ value: 'fr', label: 'french' },
];

const getLanguageOption = (language) => {
return languageOptions.findIndex(object => object.value === language);
}

console.log(getLanguageOption('en'))

File: packages/webapp/src/components/CreateUserAccount/index.jsx (hunk 2 above)
Comment (seinnlee) [reply packages/webapp/src/components/CreateUserAccount/index.jsx:81]: That's a good idea. But litefarm_lang is used throughout the app (in containers and components of webapp) and we do not have a constants file at the head of the directory. Should we create it?

**Review by anant00729 — APPROVED** [review]

**Review by smattingly — APPROVED** [review]

**Pull request merged**